### ISSUE TITLE
There will be PDFs

### ISSUE THREAD

**Issue by jennybc** [posit-dev/positron#4201]
Born out of a recent internal Slack discussion.

If you click on a PDF in the file explorer, you'll see something like this:

<img width="709" alt="Screenshot 2024-07-31 at 1 21 39 PM" src="https://github.com/user-attachments/assets/c2bbd31e-db5d-4b24-8a00-0dabf28428f2">

We think doing something more sensible with PDFs, by default, is consistent with Positron being an IDE for data science.

Possible approaches:

* Recommend or bundle an extension, such as https://open-vsx.org/extension/tomoki1207/pdf.
* Incorporate our own tooling around PDF.js (https://mozilla.github.io/pdf.js/) or equivalent. PDF.js is notably used by the extension linked above and by RStudio. @jmcphers noted we would want a PDF to open in the Viewer instead of, say, an editor tab.

**Comment by dfalbel** [comment]
I think it would be nice to have also have a way to open `xlsx` files. Even if it was 'open with whatever the system uses by default'. Currently we also get: 
 
![image](https://github.com/user-attachments/assets/1781b640-21fa-428f-a4ee-0765d650c5a6)

**Comment by nstrayer** [comment]
I'm pro rolling-our-own here for better viewer support etc..

**Comment by kylebutts** [comment]
In my opinion, I want the option to not open a pdf in the viewer. E.g., what if I wanted to compare my paper with some notes or another paper; in the editor, I can open in split tabs. In Viewer, I would not be able to

**Comment by danieltomasz** [comment]
I installed  specific extension/plugin  for having this option to open it as a tab in the editor; a separate light plugin for preview  (and edit) xlsx/csv/ tsv would be great though

**Comment by juliasilge** [comment]
If we don't support PDFs in the Viewer right away, we could bundle a PDF extension for the short term.

**Comment by michelleg06** [comment]
Has there been any progress on this?

**Comment by juliasilge** [comment]
@michelleg06 For now, we would recommend you use an extension such as https://open-vsx.org/extension/tomoki1207/pdf

**Comment by mine-cetinkaya-rundel** [comment]
With `2025.07.0 (Universal) build 134`, clicking on a PDF results in the following instead of the view @jennybc shared in the original post. I don't think this is ideal either, and I'd much prefer it opened and wouldn't care whether it's in the viewer or the editor, but wanted to share the current, different result nonetheless.

<img width="1624" alt="Image" src="https://github.com/user-attachments/assets/910c373f-fb66-43ae-b005-22e85c445e17" />

**Comment by juliasilge** [comment]
Requested again in https://github.com/posit-dev/positron/discussions/8812

**Comment by jmcphers** [comment]
@kmasiello discovered that the most popular extension for viewing PDFs (the one linked above) does not work in Posit Workbench since it's marked for desktop only.

https://positpbc.slack.com/archives/C01P9E05Y85/p1758729616510619

> This extension is disabled because it is defined to run only in Code - OSS with Posit Workbench for the Desktop

Marking for re-triage -- do we need to get to this sooner since it leaves Workbench users w/o a viable path to view PDFs?

**Comment by zeigerpuppy** [comment]
Agreed, a good PDF viewer that renders to the preview pane is essential for scientific use.
Especially as this is the expectation set in the docs: https://quarto.org/docs/get-started/hello/positron.html#rendering

The plugin at https://open-vsx.org/extension/tomoki1207/pdf does load for me in editor tab (Positron 2025.11.0 build 234).
However, previewing a .qmd file to PDF causes the PDF tab to close (even if pinned), requiring manual re-opening.

**Comment by softwarenerd** [comment]
This issue is addressed in https://github.com/posit-dev/positron/pull/11761.

**Comment by rodrigosf672** [comment]
### Verified Fixed
    Positron Version(s) : 2026.03.0 build 89
    OS Version          : macOS Tahoe 26.2

### Test scenario(s) 

#### Opening PDF behavior

| Action | Expected Result |
|---|---|
Click a .pdf in file explorer | Opens in Viewer panel not editor |
Double click a .pdf | Same as single click |
Open another PDF | Viewer content replaces previous |
Open PDF while notebook open | Notebook tab unchanged |
Reopen same PDF | No duplicate tab or reload flicker |

#### Quarto-to-PDF workflows

| Action | Expected Result |
|---|---|
Render .qmd to PDF | PDF appears in Viewer |
Re render same .qmd | Viewer refreshes existing file |
Pin PDF then re render | Tab stays and updates |
Render different .qmd | Viewer switches file |
Close viewer then render | Viewer reopens automatically |

#### Navigation and IDE interaction

| Action | Expected Result |
|---|---|
Scroll PDF then change tab | Scroll position preserved |
Click PDF then run console command | Console focus works normally |
Run notebook cell while PDF open | Execution unaffected |
Split layout and open PDF | Layout preserved |
Resize panels | PDF reflows correctly |

#### Refresh and file changes

| Action | Expected Result |
|---|---|
Modify PDF externally | Viewer refreshes or prompts |
Delete open PDF | Missing file state shown no crash |
Rename open PDF | Path updates correctly |
Git checkout changes PDF | Viewer reloads new version |

#### Multi file research workflows

| Action | Expected Result |
|---|---|
Open two PDFs side by side | Both visible independently |
Compare PDF and text notes | Editor and viewer coexist |
Switch kernels while open | PDF unaffected |
Restart session | PDF restored on reload |

#### Failure and edge cases

| Action | Expected Result |
|---|---|
Open corrupted PDF | Graceful error UI |
Open large PDF | Loads or shows progress no freeze |
Open password protected PDF | Informative unsupported message |
Offline session | Local cached file still opens |

<img width="1060" height="240" alt="Image" src="https://github.com/user-attachments/assets/9caddc3d-35d8-4ac3-a688-b7b9bc6cebf3" />
